src/main/java/com/akademia/educators_back/service/impl/ProblemService.java

```diff
@@ -32,4 +38,20 @@ public void updateProblem(ProblemTo problemTo) {
         problemEntity.setCategoryEntity(problemTo.getCategoryEntity());
         problemRepository.save(problemEntity);
     }
+
+    @Override
+    public List<ProblemTo> getProblems() {
+        List<ProblemTo> problemsTo = new ArrayList<>();
+        List<ProblemEntity> problemsEntity = problemRepository.findAll();
+        for (ProblemEntity problem : problemsEntity){
+            problemsTo.add(problemMapper.toProblemTO(problem));
+        }
+        return problemsTo;
+    }
+
+    @Override
+    public ProblemTo getProblemById(Long id) {
+        ProblemEntity problemEntity = problemRepository.findById(id).get();
```

Comment:
a jeśli nic nie znajdziemy? wywoływanie get'a na optionalu jest nieprawidłowe.
1) najpierw dajemy isPresent() i wtedy get jeśli coś jest - ale to że jest uważane za antypattern dlatego odsyłam do nr2
2) .orElseThrow( ()-> exception) i to samo w metodzie wyżej gdzie robimy findAll